luaui/Widgets/gui_unit_market.lua

```diff
@@ -0,0 +1,480 @@
+        if (#selectedUnits <= 0) then
+            return false
+        end
+        -- pressing button element
+        if mx > uiElementRect[1] and mx < uiElementRect[3] and my > uiElementRect[2] and my < uiElementRect[4] then
+            -- pressing actual button
+            if mx > buttonRect[1] and mx < buttonRect[3] and my > buttonRect[2] and my < buttonRect[4] then
+                toggleSelectedUnitsForSale(selectedUnits)
+                return true
+            end
+        end
+    end
+end
+
+local spIsGUIHidden = Spring.IsGUIHidden
+local animationDuration = 7
+local animationFrequency = 3
+function widget:DrawScreen()
+    if spIsGUIHidden() then
+        return
+    end
+    local selectedUnits = spGetSelectedUnits()
+    if (#selectedUnits <= 0) then
+        return
+    end
+
+    local alt, ctrl, meta, shift = spGetModKeyState()
+    gl.DeleteList(buttonList)
+    --gl.Color(1, 0.5, 0, 0.8) 
+    local color = {1, 0.65, 0, 0.8} -- Orange color for button background
+    local mult = 0.15
+    local x, y = spGetMouseState()
+	uiElementRect = { buttonX - (buttonW / 2) - uiPadding, buttonY - (buttonH / 2) - uiPadding, buttonX + (buttonW / 2) + uiPadding, buttonY + (buttonH / 2) + uiPadding }
+	buttonRect = { buttonX - (buttonW / 2), buttonY - (buttonH / 2), buttonX + (buttonW / 2), buttonY + (buttonH / 2) }
+    if not (alt or ctrl or shift) then
+        if x > buttonRect[1] and x < buttonRect[3] and y > buttonRect[2] and y < buttonRect[4] then
+            mult = 0.55
+        end
+    else
+        mult = 0.01
+    end
+    buttonList = gl.CreateList(function()
+		UiElement(uiElementRect[1], uiElementRect[2], uiElementRect[3], uiElementRect[4], 1, 1, 1, 1, 1, 1, 1, 1)
+        UiButton(buttonRect[1], buttonRect[2], buttonRect[3], buttonRect[4], 1, 1, 1, 1, 1, 1, 1, 1, nil, { color[1]*mult, color[2]*mult, color[3]*mult, 1 }, { color[1], color[2], color[3], 0.2 })
+    end)
+    gl.CallList(buttonList)
+
+    gl.Color(1, 1, 1, 1)  -- White color for text
+    gl.Text(sellUnitText, buttonRect[1]+((buttonRect[3]-buttonRect[1])/2), (buttonRect[2]+((buttonRect[4]-buttonRect[2])/2)) - (buttonH * 0.16), textSize * uiScale, "co")
+end
+function widget:DrawWorld()
+	if spIsGUIHidden() or next(unitsForSale) == nil then
+		return
+	end
+
+	local cameraState = spGetCameraState()
+	local camHeight = cameraState and cameraState.dist or nil
+
+	if camHeight > 9000 then
+		return
+	end
+
+	for unitID, _ in pairs(unitsForSale) do
+		local x, y, z = spGetUnitPosition(unitID)
+
+		if spIsUnitInView(unitID) and x then
+            local currentTime = spGetGameSeconds() % animationDuration
+            local animationProgress = math.sin((currentTime / animationDuration) * (2 * math.pi * animationFrequency))
+
+            local greenColorA	= {0.3, 1.0, 0.3, 1.0}
+            local redColor = 1
+            local greenColor = (0.8 + animationProgress * 0.2)
+
+            local radiusSize = 15 + animationProgress * 10
```

Comment:
Not a blocker by any means, but I think it would be nice to scale this circle based on unit size, `gui_team_platter` has patterns that can be followed for that. But that can definitely be done later if you would prefer.